internal/querynodev2/handlers.go

```diff
@@ -249,6 +251,72 @@ func (node *QueryNode) querySegments(ctx context.Context, req *querypb.QueryRequ
 	}, nil
 }
 
+func (node *QueryNode) optimizeSearchParams(ctx context.Context, req *querypb.SearchRequest, channel string) *querypb.SearchRequest {
```

Comment:
We need an `Optimizer` type, to make criterion and unify the interfaces

Reply:
So far the Optimizer signature is not clear. Will add abstraction when we have more than one optimizer here